liferay-staging/liferay-portal
# LPS-169003 Migrate Staging tree to new ClayTreeView component

Hi guys! 😄 I'm leaving this PR that migrates Staging tree view to new Clay component. Please tell me if you need any change, thanks!

Jira tickets
=======
Task: https://issues.liferay.com/browse/LPS-169001
Subtask: https://issues.liferay.com/browse/LPS-169003

Motivation
=======
The purpose of this task is migrating Staging tree to new ClayTreeView component. 

Additional information
========
- The new tree view is accesible and keyboard navigable. Use tab to move between tree items, space to check/uncheck items and left and right arrows to expand/collapse them
- In future subtasks, we will remove no needed code (https://issues.liferay.com/browse/LPS-169395) and feature flag (https://issues.liferay.com/browse/LPS-169004)
- We already passed tests in our repo (https://github.com/liferay-echo/liferay-portal/pull/10680) and our QAs think failures are not related to our changes

Steps to Verify
========
1. Enable feature flag: `feature.flag.LPS-169001=true`
2. There are five usages of this tree, this is the list and how to find them in portal:
- `export-import-web/export_layouts.jsp`: Publishing > Exports > + button
- `export-import-web/edit_publish_configuration.jsp`: Create a page > Navigate to view mode of the page  > Publish to Live > Switch to Advanced Publish Process
- `export-import-web/edit_export_configuration.jsp`: Publishing > Exports > Kebab menu > Export Templates > + button
- `staging-processes-web/publish_layouts.jsp`: Publishing > Staging > + button
- `staging-processes-web/edit_publish_configuration.jsp`: Publishing > Staging > Kebab menu > Publish Templates
4. Check the tree works as expected

Screenshots
========
![image](https://user-images.githubusercontent.com/2444936/209142366-34670def-46d9-4253-a829-c55f24bdf707.png)
